docs/RFCS/select_from_index.md

```diff
@@ -0,0 +1,55 @@
+- Feature Name: select_from_index
+- Status: draft
+- Start Date: 2015-08-10
+- RFC PR:
+- Cockroach Issue: #2046
+
+# Summary
+
+Extend the SELECT syntax to allow an optional index to be specified in
+table references. The current table reference syntax is `table` or
+`database.table`. This syntax would be extended to allow `table.index`
+and `database.table.index`.
+
+# Motivation
+
+As indexes are optional in SQL, it is difficult to know when an index
+is used to satisfy a query which in turn makes it difficult to use the
+existing logic test infrastructure to test correctness of index
+scans. Providing for specification of an index to read from will make
```

Comment:
Specifying the index will essentially make a view containing only the columns in the index with the exception that the primary-key index contains all of the columns in the table. (There is no separate index for the primary-key: the primary-key index is the table).

So in your example, `SELECT pk from tbl@primary WHERE name='Harold'` will work exactly as `SELECT pk from tbl WHERE name='HAROLD'` (note the name of the primary-key index is `primary`, not `pk`).

You're correct that we could get even more control by specifying the index to use for portions of the where-clause. While true, that is significantly larger in scope than this proposal.